internal/state/replicator.go

```diff
@@ -63,69 +64,76 @@ func (e Entity) GetTag() string {
 // Replicate replicates images from the source registry to the Zot registry.
 func (r *BasicReplicator) Replicate(ctx context.Context, replicationEntities []Entity) error {
 	log := logger.FromContext(ctx)
-	pullAuthConfig := authn.FromConfig(authn.AuthConfig{
-		Username: r.sourceUsername,
-		Password: r.sourcePassword,
-	})
-	pushAuthConfig := authn.FromConfig(authn.AuthConfig{
-		Username: r.remoteUsername,
-		Password: r.remotePassword,
-	})
-
-	pullOptions := []crane.Option{crane.WithAuth(pullAuthConfig)}
-	pushOptions := []crane.Option{crane.WithAuth(pushAuthConfig)}
-
-	if r.useUnsecure {
-		pullOptions = append(pullOptions, crane.Insecure)
-		pushOptions = append(pushOptions, crane.Insecure)
-	}
-
-	for _, replicationEntity := range replicationEntities {
+	for _, entity := range replicationEntities {
+		sourceImage := fmt.Sprintf("%s/%s:%s", r.sourceRegistry, entity.Repository, entity.Tag)
+		destinationImage := fmt.Sprintf("%s/%s:%s", r.remoteRegistryURL, entity.Repository, entity.Tag)
+
+		// Get image size before transfer
```

Comment:
It would be better to get the image size before we attempt to pull the entire image. This should be possible with crane by using `crane.Manifest()`. See: https://github.com/google/go-containerregistry/blob/main/cmd/crane/recipes.md#get-total-image-size